### PR TITLE
staticupload: don't request empty invalidation

### PR DESCRIPTION
**Pull request by derpsteb**
### Proposed change(s)

- If no files have been touched, do not initiate an invalidation. Otherwise, the AWS API is unhappy.

### CODE REVIEW

File: internal/staticupload/staticupload.go [hunk 1]
```diff
@@ -171,6 +181,10 @@ func (c *Client) invalidateCacheForKeys(ctx context.Context, keys []string) (str
 		return "", InvalidationError{inner: fmt.Errorf("too many keys to invalidate: %d", len(keys))}
 	}
 
+	if len(keys) == 0 {
+		return "", InvalidationError{inner: errors.New("no keys to invalidate")}
+	}
+
```
Comment (daniel-weisse) [inline internal/staticupload/staticupload.go:187]: I would assume the cdn client errors anyway if we pass an empty list of keys?
In that case we wouldnt need to manually return an error here. 
Of course this depends on if the error has side effects in the cdn or similar

**Review by daniel-weisse — APPROVED** [review]
Can you add tests for empty lists of invalidation keys?

**Review by malt3 — APPROVED** [review]
